frontend/src/download.js

```diff
@@ -1,14 +1,53 @@
 const FileReceiver = require('./fileReceiver');
-const { notify } = require('./utils');
+const { notify, findMetric, isFile } = require('./utils');
 const $ = require('jquery');
 require('jquery-circle-progress');
 
 const Raven = window.Raven;
+
+if (!localStorage.hasOwnProperty('totalDownloads')) {
+  localStorage.setItem('totalDownloads', 0);
+}
+
 $(document).ready(function() {
   //link back to homepage
   $('.send-new').attr('href', window.location.origin);
 
+  if (location.pathname.toString().includes('download')) {
+    $('.send-new').click(function(target) {
+      target.preventDefault();
+      window.analytics
+            .sendEvent('recipient', 'restarted', {
+              cd2: 'completed'
+            })
+            .then(() => {
+              location.href = target.currentTarget.href;
+            });
+    })
+
+
+    $('.legal-links a, .social-links a, #dl-firefox').click(function(target) {
+      target.preventDefault();
+      const metric = findMetric(target.currentTarget.href);
+      // record exited event by recipient
+      window.analytics
+            .sendEvent('recipient', 'exited', {
+              cd3: metric
+            })
+            .then(() => {
+              location.href = target.currentTarget.href;
+            });
+    })
+
+    $('#expired-send-new').click(function() {
+      localStorage.setItem('referrer', 'errored-download');
+    })
+
+  }
+
   const filename = $('#dl-filename').html();
+  const bytelength = $('#dl-bytelength').html();
```

Comment:
Changed to `const bytelength = Number($('#dl-bytelength').text());`, and the same for `timeToExpiry`.